keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/Optimizer.java

```diff
@@ -16,17 +16,48 @@
 import io.improbable.keanu.vertices.Vertex;
 
 public interface Optimizer {
-
+    /**
+     * Adds a callback to be called whenever the optimizer evaluates the fitness of a point.
+     * @param fitnessCalculationHandler a function to be called whenever the optimizer evaluates the fitness of a point.
+     *                                  The double[] argument to the handler represents the point being evaluated.
+     *                                  The Double argument to the handler represents the fitness of that point.
+     */
     void addFitnessCalculationHandler(BiConsumer<double[], Double> fitnessCalculationHandler);
 
+    /**
+     * This attempts to remove a callback function that previously would have been called whenever the optimizer
+     * evaluated the fitness of a point. If the callback is not registered then this function will do nothing.
+     * @param fitnessCalculationHandler the function to be removed from the list of fitness evaluation callbacks
+     */
     void removeFitnessCalculationHandler(BiConsumer<double[], Double> fitnessCalculationHandler);
 
+    /**
+     * This methods detects the vertices in the Bayesian Network that have been observed, and will use MAP estimation to
```

Comment:
Not sure if we need to state that the observed vertices will be detected. Also, saying MAP optimises the the observations probability of the observed vertices is misleading since MAP optimises the probability of the whole BN, including priors.